core/src/main/java/hudson/tools/ZipExtractionInstaller.java

```diff
@@ -83,8 +82,12 @@ public String getSubdir() {
     @Override
     public FilePath performInstallation(ToolInstallation tool, Node node, TaskListener log) throws IOException, InterruptedException {
         FilePath dir = preferredLocation(tool, node);
-        if (dir.installIfNecessaryFrom(new URL(url), log, "Unpacking " + url + " to " + dir + " on " + node.getDisplayName())) {
-            dir.act(new ChmodRecAPlusX());
+        try {
+            if (dir.installIfNecessaryFrom(new URI(url).toURL(), log, "Unpacking " + url + " to " + dir + " on " + node.getDisplayName())) {
+                dir.act(new ChmodRecAPlusX());
+            }
+        } catch (URISyntaxException e) {
```

Comment:
this should not swallow the exception - as the action has not performed this needs to throw an Exception.

I suggest wrapping the URISyntaxException into a MalformedURLException as done elsewhere.

Reply:
Implemented the following as per suggestion
```
catch (URISyntaxException e) {
            MalformedURLException mex = new MalformedURLException(e.getMessage());
            mex.initCause(e);
            throw mex;
```